backend/src/app/srepl/fixes.clj

```diff
@@ -279,3 +279,48 @@
 
          (when (contains? (:features file) "storage/pointer-map")
            (files/persist-pointers! h/*conn* (:id file))))))))
+
+(defn fix-main-shape-name
+  ([file]
+   (prn (str "Updating " (:name file) " " (:id file)))
+   (if-not (contains? (:features file) "components/v2")
+     (prn "   This file is not v2")
+     (let [libs (->> (files/get-file-libraries h/*conn* (:id file))
+                     (cons file)
+                     (map #(files/get-file h/*conn* (:id %) (:features file)))
+                     (d/index-by :id))
+
+           update-shape
+           (fn [shape]
+             (if-not (ctk/instance-head? shape)
+               shape
+               (let [component (ctf/get-component libs (:component-file shape) (:component-id shape) {:include-deleted? true})
+                     [_path name] (cph/parse-path-name (:name shape))
+                     full-name (cph/clean-path (str (:path component) "/" name))]
+                 (if (= name (:name component))
+                   (assoc shape :name full-name)
+                   shape))))
+
+
+           update-page
+           (fn [page]
+             (prn (str "Page " (:name page)))
+             (h/update-shapes page update-shape))]
+
+       (update file :data h/update-pages update-page))))
+
+  ([file save?]
+   (let [file (-> file
+                  (update :data blob/decode)
+                  (fix-main-shape-name))]
```

Comment:
I don't know how this fix function can be used but it has at least two flaws:
- The arity 1 of fix-main-shape-name can return nil if the file is not components-v2 that can just delete the file data
- The arity 2 handles pointers but does not setup the registry binding for delimit the pointers, so on large number of files, the last file will persist all pointers of all the other files processed.

Reply:
if this will be used with `h/update-file` then the pointer handling is already covered.
also, if you don't touch anything related to features maybe you just can skip encoding and decoding them